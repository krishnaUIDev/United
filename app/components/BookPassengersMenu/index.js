import React from 'react';
import PropTypes from 'prop-types';
import { TAB_KEY } from 'containers/App/constants';
import BookPassengersSection from '../BookPassengersSection';

const INFANT_MAX = 8;

class BookPassengersMenu extends React.Component { // eslint-disable-line
  componentDidMount() {
    // disable buttons based on groupSum
    this.disableButtons();
    for (const index in this.props.ids) { // eslint-disable-line
      const currValue = this.props.ids[index];
      const arrayName = (this.props.passengerModelObject) ? this.props.passengerModelObject[currValue] : currValue;
      this.props.onChangeModel(`${this.props.parentModel}.${arrayName}`, this.props.passengersToBook.get(currValue));
    }
  }

  onQuantityClick(event, id, operation, modelName, limit) {
    if (event) {
      event.preventDefault();
    }
    if (document.activeElement.type !== 'number') {
      let value = parseInt(this.props.passengersToBook.get(id), 10);
      const increasedInfants = this.getInfantCounts().infantsLap + this.getInfantCounts().infantsUnderTwo + 1;
      const invalidInfantEntry = (this.props.containsInfants === 'true') ? (((id === this.props.infantTitleA) || (id === this.props.infantTitleB)) && increasedInfants > 8) : false;
      if (operation === 'plus') {
        value = isNaN(value) ? 0 : value;
        const groupSum = this.getGroupSum();
        value = ((invalidInfantEntry) || (groupSum === limit) || ((this.props.individualMaxLimit) && value === this.props.individualMaxLimit)) ? value : value += 1;
      } else if (id === this.props.minValueIsOne && value === 1) {
        value = 1;
      } else {
        value = isNaN(value) ? 0 : value;
        value = (value <= 1) ? 0 : value -= 1;
      }
      this.props.onChangePassenger(id, value);
      this.props.onChangeModel(modelName, value);
      setTimeout(() => {
        this.disableButtons(modelName, operation);
      }, 1);
      return value;
    }
    return event.target.value;
  }

  onTextInput(event, id, model) {
    const keyCode = event.keyCode;
    // 38 is up, 40 is down, 9 is tab
    if (keyCode !== TAB_KEY) {
      const num = Number.parseInt((event.target.value !== '') ? event.target.value : 0, 10);
      const groupSum = (this.getGroupSum(num, id));
      let invalidInfantEntry;
      if (id === this.props.infantTitleA) {
        invalidInfantEntry = (num + this.getInfantCounts().infantsLap) > INFANT_MAX;
      } else if (id === this.props.infantTitleB) {
        invalidInfantEntry = (num + this.getInfantCounts().infantsUnderTwo) > INFANT_MAX;
      }
      // select it so that it will be replaced next time key is pressed
      if ((id === this.props.minValueIsOne) && (num === 0)) {
        this.props.onChangePassenger(id, 1);
        this.props.onChangeModel(`${this.props.parentModel}${model}`, 1);
        setTimeout(() => {
          this.disableButtons();
        }, 1);
        this.props.onQuantityInputFocus(event);
        return 1;
      } else if (num < 0 || isNaN(num)) {
        let returnValue;
        if (model === this.props.modelDefaultToOne) {
          this.props.onChangeModel(`${this.props.parentModel}${model}`, 1);
          this.props.onQuantityInputFocus(event);
          this.props.onChangePassenger(id, 1);
          returnValue = 1;
        } else {
          this.props.onChangeModel(`${this.props.parentModel}${model}`, 0);
          this.props.onQuantityInputFocus(event);
          this.props.onChangePassenger(id, 0);
          returnValue = 0;
        }
        setTimeout(() => {
          this.disableButtons();
        }, 1);
        return returnValue;
      } else if ((invalidInfantEntry) || (groupSum) > this.props.maxLimit || ((this.props.individualMaxLimit) && num > (this.props.individualMaxLimit))) {
        // set it back to original value before entering input
        const originalVal = this.props.passengersToBook.get(id);
        this.props.onChangeModel(`${this.props.parentModel}${model}`, originalVal);
        this.props.onQuantityInputFocus(event);
        setTimeout(() => {
          this.disableButtons();
        }, 1);
        return originalVal;
      }
      this.props.onChangePassenger(id, num);
      this.props.onChangeModel(`${this.props.parentModel}${model}`, num);
      setTimeout(() => {
        this.disableButtons();
      }, 1);
      this.props.onQuantityInputFocus(event);
      return num;
    }
    return event;
  }

  getInfantCounts() {
    const infantsA = this.props.passengersToBook.get(this.props.infantTitleB);
    const infantsB = this.props.passengersToBook.get(this.props.infantTitleA);
    return { infantsLap: infantsA, infantsUnderTwo: infantsB };
  }

  getGroupSum(number, id) {
    let groupSum = 0;
    if (number && id) {
      for (const index in this.props.fullArray) { // eslint-disable-line
        const currValue = this.props.fullArray[index];
        if (currValue !== id) {
          groupSum += parseInt(this.props.passengersToBook.get(currValue), 10);
        }
      }
      groupSum += number;
    } else {
      for (const index in this.props.fullArray) { // eslint-disable-line
        const currValue = this.props.fullArray[index];
        groupSum += parseInt(this.props.passengersToBook.get(currValue), 10);
      }
    }
    return groupSum;
  }

  getPassengersData(minusBtnEnabled, plusBtnEnabled, passengersToBook) {
    const remaining = (this.props.maxLimit - this.getGroupSum());
    const bookPassengersData = this.props.arrayIndexs.map((index) =>
      <BookPassengersSection
        id={this.props.ids[index]}
        key={index}
        index={index}
        sectionTitle={this.props.sectionTitle[index]}
        sectionDetail={this.props.sectionDetail[index]}
        maxLimit={this.props.maxLimit}
        maxRemaining={remaining}
        onTextInput={(event, id, model) => this.onTextInput(event, id, model)}
        onQuantityClick={(event, id, operation, limit) => this.onQuantityClick(event, id, operation, limit)}
        minusBtnDisabled={minusBtnEnabled}
        plusBtnDisabled={plusBtnEnabled}
        currentAmount={passengersToBook.get(this.props.ids[index])}
        onQuantityInputFocus={(event) => this.props.onQuantityInputFocus(event)}
        decrementTabIndex={this.props.decrementTabIndex}
        numberOfTravelersTotal={this.props.numberOfTravelersTotal}
        parentModel={this.props.parentModel}
        modelName={this.props.parentModel === 'bookHotelModel' ? `${this.props.parentModel}.['${this.props.ids[index]}']` : `booking.passengersToBook['${this.props.ids[index]}']`}
        disableIndividualSections={this.props.disableIndividualSections}
        individualMaxLimit={this.props.individualMaxLimit}
        minValueIsOne={this.props.minValueIsOne}
      />
    );
    return bookPassengersData;
  }

  disableButtons(model, operation) {
    let disablePlusBtn;
    let disableMinusBtn;
    let disabled = false;
    if (this.getGroupSum() === 0) {
      disablePlusBtn = false;
      disableMinusBtn = true;
      disabled = true;
    } else if (this.getGroupSum() === this.props.maxLimit) {
      disablePlusBtn = true;
      disableMinusBtn = false;
      disabled = true;
    } else {
      disablePlusBtn = false;
      disableMinusBtn = false;
    }
    this.props.onDisableTravelerButtons('plusBtn', disablePlusBtn);
    this.props.onDisableTravelerButtons('minusBtn', disableMinusBtn);
    if (((operation === 'minus') && (disableMinusBtn)) || ((operation === 'plus') && (disablePlusBtn))) {
      setTimeout(() => {
        this.props.onFocusModel(`${this.props.parentModel}${model}`);
      }, 1);
    }
    return disabled;
  }

  render() {
    const showMenu = (this.props.showTravelerMenu) ? 'block' : 'none';
    const passengersToBook = this.props.passengersToBook;
    const minusBtnEnabled = (this.props.toDisableBtns) ? this.props.toDisableBtns.get('minusBtn') : '';
    const plusBtnEnabled = (this.props.toDisableBtns) ? this.props.toDisableBtns.get('plusBtn') : '';

    const bookPassengersData = this.getPassengersData(minusBtnEnabled, plusBtnEnabled, passengersToBook);

    return (
      <div
        aria-label="Passengers"
        id="incrementerContainer"
        style={{ display: showMenu }}
      >
        {bookPassengersData}
      </div>
    );
  }
}

BookPassengersMenu.propTypes = {
  sectionTitle: PropTypes.array,
  sectionDetail: PropTypes.array,
  ids: PropTypes.array,
  fullArray: PropTypes.array,
  maxLimit: PropTypes.number,
  onChangePassenger: PropTypes.func,
  showTravelerMenu: PropTypes.bool,
  passengersToBook: PropTypes.object,
  onDisableTravelerButtons: PropTypes.func,
  toDisableBtns: PropTypes.object,
  onQuantityInputFocus: PropTypes.func,
  decrementTabIndex: PropTypes.number,
  arrayIndexs: PropTypes.array,
  onChangeModel: PropTypes.func,
  onFocusModel: PropTypes.func,
  parentModel: PropTypes.string,
  passengerModelObject: PropTypes.object,
  containsInfants: PropTypes.string,
  infantTitleA: PropTypes.string,
  infantTitleB: PropTypes.string,
  modelDefaultToOne: PropTypes.string,
  numberOfTravelersTotal: PropTypes.string,
  individualMaxLimit: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  minValueIsOne: PropTypes.string,
  disableIndividualSections: PropTypes.string,
};

export default BookPassengersMenu;
