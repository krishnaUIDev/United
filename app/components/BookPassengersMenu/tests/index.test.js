import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import Immutable from 'immutable';

import BookPassengersMenu from '../index';

describe('<BookPassengersMenu />', () => {
  const bookPassSpy = spy();
  const changeSpy = spy();
  const inputSpy = spy();
  const passToBook = Immutable.Map({
    'Adults (18-64)': 2,
    'Seniors (65+)': 0,
    'Infants (under 2)': 3,
    'Infants on lap': 1,
    'Children (15-17)': 0,
    'Children (12-14)': 0,
    'Children (5-11)': 0,
    'Children (2-4)': 0,
  });
  const renderedComponent = shallow(
    <BookPassengersMenu
      ids={['Adults (18-64)', 'Seniors (65+)', 'Infants (under 2)', 'Infants on lap']}
      sectionTitle={['Adults', 'Seniors', 'Infants', 'Infants on lap']}
      sectionDetail={['(18-64)', '(65+)', '(under 2)', '']}
      fullArray={['Seniors (65+)', 'Infants (under 2)', 'Infants on lap']}
      onQuantityClick={bookPassSpy}
      toDisableBtns={Immutable.Map({ plusBtn: false, minusBtn: false })}
      passengersToBook={passToBook}
      onDisableTravelerButtons={bookPassSpy}
      onChangePassenger={bookPassSpy}
      arrayIndexs={['0', '1', '2', '3']}
      onChangeModel={changeSpy}
      onQuantityInputFocus={inputSpy}
      infantTitleA="Infants (under 2)"
      infantTitleB="Infants on lap"
    />
  );
  const wrapper = renderedComponent.instance();
  it('render BookPassengersMenu Component', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
  it('render component prop infantTitleA', () => {
    expect(wrapper.props.infantTitleA).toEqual('Infants (under 2)');
  });
  it('render component prop infantTitleB', () => {
    expect(wrapper.props.infantTitleB).toEqual('Infants on lap');
  });
  it('should render divs', () => {
    expect(renderedComponent.find('div').length).toEqual(1);
  });
  it('should render BookPassengersSection', () => {
    expect(renderedComponent.find('BookPassengersSection').length).toEqual(4);
  });
  it('should handle disableButtons()', () => {
    expect(renderedComponent.instance().disableButtons()).toEqual(false);
    expect(bookPassSpy.called).toEqual(true);
  });
  it('should handle getGroupSum()', () => {
    expect(renderedComponent.instance().getGroupSum()).toEqual(4);
  });
  it('should handle getGroupSum() with parameters', () => {
    expect(renderedComponent.instance().getGroupSum(3, 'Adults (18-64)')).toEqual(7);
  });
  it('should handle onQuantityClick()', () => {
    expect(renderedComponent.instance().onQuantityClick(null, 'Adults (18-64)', 'plus', 3)).toEqual(3);
    expect(bookPassSpy.called).toEqual(true);
    expect(changeSpy.called).toEqual(true);
  });
  it('should handle onTextInput()', () => {
    expect(renderedComponent.instance().onTextInput({ keyCode: 51, target: { value: '7' } }, 'Adults (18-64)')).toEqual(7);
    expect(bookPassSpy.called).toEqual(true);
    expect(changeSpy.called).toEqual(true);
    expect(inputSpy.called).toEqual(true);
  });
  it('should render BookPassengersSection props', () => {
    const section = renderedComponent.find('BookPassengersSection').node.props;
    expect(section.id).toEqual('Adults (18-64)');
    expect(section.index).toEqual('0');
    expect(section.sectionTitle).toEqual('Adults');
    expect(section.sectionDetail).toEqual('(18-64)');
    expect(section.onTextInput).toBeDefined();
    expect(typeof (section.onTextInput)).toEqual('function');
    expect(section.onQuantityClick).toBeDefined();
    expect(typeof (section.onQuantityClick)).toEqual('function');
    expect(section.onQuantityInputFocus).toBeDefined();
    expect(typeof (section.onQuantityInputFocus)).toEqual('function');
    expect(section.minusBtnDisabled).toEqual(false);
    expect(section.plusBtnDisabled).toEqual(false);
  });
  it('should handle getInfantCounts()', () => {
    expect(renderedComponent.instance().getInfantCounts()).toEqual({ infantsLap: 1, infantsUnderTwo: 3 });
  });
  describe('should handle getPassengersData()', () => {
    const data = renderedComponent.instance().getPassengersData(false, false, passToBook);
    it('should render Adults (18-64)', () => {
      const adults = data[0];
      expect(adults.type).toBeDefined();
      expect(typeof (adults.type)).toEqual('function');
      expect(adults.ref).toEqual(null);
      expect(adults.key).toEqual('0');
      const props = adults.props;
      expect(props.id).toEqual('Adults (18-64)');
      expect(props.index).toEqual('0');
      expect(props.sectionTitle).toEqual('Adults');
      expect(props.sectionDetail).toEqual('(18-64)');
      expect(props.minusBtnDisabled).toEqual(false);
      expect(props.plusBtnDisabled).toEqual(false);
      expect(props.currentAmount).toEqual(2);
    });
    it('should render Seniors (65+)', () => {
      const seniors = data[1];
      expect(seniors.type).toBeDefined();
      expect(typeof (seniors.type)).toEqual('function');
      expect(seniors.ref).toEqual(null);
      expect(seniors.key).toEqual('1');
      const props = seniors.props;
      expect(props.id).toEqual('Seniors (65+)');
      expect(props.index).toEqual('1');
      expect(props.sectionTitle).toEqual('Seniors');
      expect(props.sectionDetail).toEqual('(65+)');
      expect(props.minusBtnDisabled).toEqual(false);
      expect(props.plusBtnDisabled).toEqual(false);
      expect(props.currentAmount).toEqual(0);
    });
    it('should render Infants (under 2)', () => {
      const infants = data[2];
      expect(infants.type).toBeDefined();
      expect(typeof (infants.type)).toEqual('function');
      expect(infants.ref).toEqual(null);
      expect(infants.key).toEqual('2');
      const props = infants.props;
      expect(props.id).toEqual('Infants (under 2)');
      expect(props.index).toEqual('2');
      expect(props.sectionTitle).toEqual('Infants');
      expect(props.sectionDetail).toEqual('(under 2)');
      expect(props.minusBtnDisabled).toEqual(false);
      expect(props.plusBtnDisabled).toEqual(false);
      expect(props.currentAmount).toEqual(3);
    });
    it('should render Infants on lap', () => {
      const infants = data[3];
      expect(infants.type).toBeDefined();
      expect(typeof (infants.type)).toEqual('function');
      expect(infants.ref).toEqual(null);
      expect(infants.key).toEqual('3');
      const props = infants.props;
      expect(props.id).toEqual('Infants on lap');
      expect(props.index).toEqual('3');
      expect(props.sectionTitle).toEqual('Infants on lap');
      expect(props.sectionDetail).toEqual('');
      expect(props.minusBtnDisabled).toEqual(false);
      expect(props.plusBtnDisabled).toEqual(false);
      expect(props.currentAmount).toEqual(1);
    });
  });
});
