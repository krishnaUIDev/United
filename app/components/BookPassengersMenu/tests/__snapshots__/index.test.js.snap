// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookPassengersMenu /> render BookPassengersMenu Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    aria-label="Passengers"
    id="incrementerContainer"
    style={
        Object {
            "display": "none",
          }
    }
>
    <BookPassengersSection
        currentAmount={2}
        decrementTabIndex={undefined}
        disableIndividualSections={undefined}
        id="Adults (18-64)"
        index="0"
        individualMaxLimit={undefined}
        maxLimit={undefined}
        maxRemaining={NaN}
        minValueIsOne={undefined}
        minusBtnDisabled={false}
        modelName="booking.passengersToBook['Adults (18-64)']"
        numberOfTravelersTotal={undefined}
        onQuantityClick={[Function]}
        onQuantityInputFocus={[Function]}
        onTextInput={[Function]}
        parentModel={undefined}
        plusBtnDisabled={false}
        sectionDetail="(18-64)"
        sectionTitle="Adults"
    />
    <BookPassengersSection
        currentAmount={0}
        decrementTabIndex={undefined}
        disableIndividualSections={undefined}
        id="Seniors (65+)"
        index="1"
        individualMaxLimit={undefined}
        maxLimit={undefined}
        maxRemaining={NaN}
        minValueIsOne={undefined}
        minusBtnDisabled={false}
        modelName="booking.passengersToBook['Seniors (65+)']"
        numberOfTravelersTotal={undefined}
        onQuantityClick={[Function]}
        onQuantityInputFocus={[Function]}
        onTextInput={[Function]}
        parentModel={undefined}
        plusBtnDisabled={false}
        sectionDetail="(65+)"
        sectionTitle="Seniors"
    />
    <BookPassengersSection
        currentAmount={3}
        decrementTabIndex={undefined}
        disableIndividualSections={undefined}
        id="Infants (under 2)"
        index="2"
        individualMaxLimit={undefined}
        maxLimit={undefined}
        maxRemaining={NaN}
        minValueIsOne={undefined}
        minusBtnDisabled={false}
        modelName="booking.passengersToBook['Infants (under 2)']"
        numberOfTravelersTotal={undefined}
        onQuantityClick={[Function]}
        onQuantityInputFocus={[Function]}
        onTextInput={[Function]}
        parentModel={undefined}
        plusBtnDisabled={false}
        sectionDetail="(under 2)"
        sectionTitle="Infants"
    />
    <BookPassengersSection
        currentAmount={1}
        decrementTabIndex={undefined}
        disableIndividualSections={undefined}
        id="Infants on lap"
        index="3"
        individualMaxLimit={undefined}
        maxLimit={undefined}
        maxRemaining={NaN}
        minValueIsOne={undefined}
        minusBtnDisabled={false}
        modelName="booking.passengersToBook['Infants on lap']"
        numberOfTravelersTotal={undefined}
        onQuantityClick={[Function]}
        onQuantityInputFocus={[Function]}
        onTextInput={[Function]}
        parentModel={undefined}
        plusBtnDisabled={false}
        sectionDetail=""
        sectionTitle="Infants on lap"
    />
</div>,
  "nodes": Array [
    <div
      aria-label="Passengers"
      id="incrementerContainer"
      style={
            Object {
                  "display": "none",
                }
      }
>
      <BookPassengersSection
            currentAmount={2}
            decrementTabIndex={undefined}
            disableIndividualSections={undefined}
            id="Adults (18-64)"
            index="0"
            individualMaxLimit={undefined}
            maxLimit={undefined}
            maxRemaining={NaN}
            minValueIsOne={undefined}
            minusBtnDisabled={false}
            modelName="booking.passengersToBook['Adults (18-64)']"
            numberOfTravelersTotal={undefined}
            onQuantityClick={[Function]}
            onQuantityInputFocus={[Function]}
            onTextInput={[Function]}
            parentModel={undefined}
            plusBtnDisabled={false}
            sectionDetail="(18-64)"
            sectionTitle="Adults"
      />
      <BookPassengersSection
            currentAmount={0}
            decrementTabIndex={undefined}
            disableIndividualSections={undefined}
            id="Seniors (65+)"
            index="1"
            individualMaxLimit={undefined}
            maxLimit={undefined}
            maxRemaining={NaN}
            minValueIsOne={undefined}
            minusBtnDisabled={false}
            modelName="booking.passengersToBook['Seniors (65+)']"
            numberOfTravelersTotal={undefined}
            onQuantityClick={[Function]}
            onQuantityInputFocus={[Function]}
            onTextInput={[Function]}
            parentModel={undefined}
            plusBtnDisabled={false}
            sectionDetail="(65+)"
            sectionTitle="Seniors"
      />
      <BookPassengersSection
            currentAmount={3}
            decrementTabIndex={undefined}
            disableIndividualSections={undefined}
            id="Infants (under 2)"
            index="2"
            individualMaxLimit={undefined}
            maxLimit={undefined}
            maxRemaining={NaN}
            minValueIsOne={undefined}
            minusBtnDisabled={false}
            modelName="booking.passengersToBook['Infants (under 2)']"
            numberOfTravelersTotal={undefined}
            onQuantityClick={[Function]}
            onQuantityInputFocus={[Function]}
            onTextInput={[Function]}
            parentModel={undefined}
            plusBtnDisabled={false}
            sectionDetail="(under 2)"
            sectionTitle="Infants"
      />
      <BookPassengersSection
            currentAmount={1}
            decrementTabIndex={undefined}
            disableIndividualSections={undefined}
            id="Infants on lap"
            index="3"
            individualMaxLimit={undefined}
            maxLimit={undefined}
            maxRemaining={NaN}
            minValueIsOne={undefined}
            minusBtnDisabled={false}
            modelName="booking.passengersToBook['Infants on lap']"
            numberOfTravelersTotal={undefined}
            onQuantityClick={[Function]}
            onQuantityInputFocus={[Function]}
            onTextInput={[Function]}
            parentModel={undefined}
            plusBtnDisabled={false}
            sectionDetail=""
            sectionTitle="Infants on lap"
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <BookPassengersMenu
        arrayIndexs={
                Array [
                        "0",
                        "1",
                        "2",
                        "3",
                      ]
        }
        fullArray={
                Array [
                        "Seniors (65+)",
                        "Infants (under 2)",
                        "Infants on lap",
                      ]
        }
        ids={
                Array [
                        "Adults (18-64)",
                        "Seniors (65+)",
                        "Infants (under 2)",
                        "Infants on lap",
                      ]
        }
        infantTitleA="Infants (under 2)"
        infantTitleB="Infants on lap"
        onChangeModel={[Function]}
        onChangePassenger={[Function]}
        onDisableTravelerButtons={[Function]}
        onQuantityClick={[Function]}
        onQuantityInputFocus={[Function]}
        passengersToBook={
                Immutable.Map {
                        Adults (18-64): 2,
                        Seniors (65+): 0,
                        Infants (under 2): 3,
                        Infants on lap: 1,
                        Children (15-17): 0,
                        Children (12-14): 0,
                        Children (5-11): 0,
                        Children (2-4): 0,
                }
        }
        sectionDetail={
                Array [
                        "(18-64)",
                        "(65+)",
                        "(under 2)",
                        "",
                      ]
        }
        sectionTitle={
                Array [
                        "Adults",
                        "Seniors",
                        "Infants",
                        "Infants on lap",
                      ]
        }
        toDisableBtns={
                Immutable.Map {
                        plusBtn: false,
                        minusBtn: false,
                }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": BookPassengersMenu {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "arrayIndexs": Array [
            "0",
            "1",
            "2",
            "3",
          ],
          "fullArray": Array [
            "Seniors (65+)",
            "Infants (under 2)",
            "Infants on lap",
          ],
          "ids": Array [
            "Adults (18-64)",
            "Seniors (65+)",
            "Infants (under 2)",
            "Infants on lap",
          ],
          "infantTitleA": "Infants (under 2)",
          "infantTitleB": "Infants on lap",
          "onChangeModel": [Function],
          "onChangePassenger": [Function],
          "onDisableTravelerButtons": [Function],
          "onQuantityClick": [Function],
          "onQuantityInputFocus": [Function],
          "passengersToBook": Immutable.Map {
            Adults (18-64): 2,
            Seniors (65+): 0,
            Infants (under 2): 3,
            Infants on lap: 1,
            Children (15-17): 0,
            Children (12-14): 0,
            Children (5-11): 0,
            Children (2-4): 0,
},
          "sectionDetail": Array [
            "(18-64)",
            "(65+)",
            "(under 2)",
            "",
          ],
          "sectionTitle": Array [
            "Adults",
            "Seniors",
            "Infants",
            "Infants on lap",
          ],
          "toDisableBtns": Immutable.Map {
            plusBtn: false,
            minusBtn: false,
},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          aria-label="Passengers"
          id="incrementerContainer"
          style={
                    Object {
                              "display": "none",
                            }
          }
>
          <BookPassengersSection
                    currentAmount={2}
                    decrementTabIndex={undefined}
                    disableIndividualSections={undefined}
                    id="Adults (18-64)"
                    index="0"
                    individualMaxLimit={undefined}
                    maxLimit={undefined}
                    maxRemaining={NaN}
                    minValueIsOne={undefined}
                    minusBtnDisabled={false}
                    modelName="booking.passengersToBook['Adults (18-64)']"
                    numberOfTravelersTotal={undefined}
                    onQuantityClick={[Function]}
                    onQuantityInputFocus={[Function]}
                    onTextInput={[Function]}
                    parentModel={undefined}
                    plusBtnDisabled={false}
                    sectionDetail="(18-64)"
                    sectionTitle="Adults"
          />
          <BookPassengersSection
                    currentAmount={0}
                    decrementTabIndex={undefined}
                    disableIndividualSections={undefined}
                    id="Seniors (65+)"
                    index="1"
                    individualMaxLimit={undefined}
                    maxLimit={undefined}
                    maxRemaining={NaN}
                    minValueIsOne={undefined}
                    minusBtnDisabled={false}
                    modelName="booking.passengersToBook['Seniors (65+)']"
                    numberOfTravelersTotal={undefined}
                    onQuantityClick={[Function]}
                    onQuantityInputFocus={[Function]}
                    onTextInput={[Function]}
                    parentModel={undefined}
                    plusBtnDisabled={false}
                    sectionDetail="(65+)"
                    sectionTitle="Seniors"
          />
          <BookPassengersSection
                    currentAmount={3}
                    decrementTabIndex={undefined}
                    disableIndividualSections={undefined}
                    id="Infants (under 2)"
                    index="2"
                    individualMaxLimit={undefined}
                    maxLimit={undefined}
                    maxRemaining={NaN}
                    minValueIsOne={undefined}
                    minusBtnDisabled={false}
                    modelName="booking.passengersToBook['Infants (under 2)']"
                    numberOfTravelersTotal={undefined}
                    onQuantityClick={[Function]}
                    onQuantityInputFocus={[Function]}
                    onTextInput={[Function]}
                    parentModel={undefined}
                    plusBtnDisabled={false}
                    sectionDetail="(under 2)"
                    sectionTitle="Infants"
          />
          <BookPassengersSection
                    currentAmount={1}
                    decrementTabIndex={undefined}
                    disableIndividualSections={undefined}
                    id="Infants on lap"
                    index="3"
                    individualMaxLimit={undefined}
                    maxLimit={undefined}
                    maxRemaining={NaN}
                    minValueIsOne={undefined}
                    minusBtnDisabled={false}
                    modelName="booking.passengersToBook['Infants on lap']"
                    numberOfTravelersTotal={undefined}
                    onQuantityClick={[Function]}
                    onQuantityInputFocus={[Function]}
                    onTextInput={[Function]}
                    parentModel={undefined}
                    plusBtnDisabled={false}
                    sectionDetail=""
                    sectionTitle="Infants on lap"
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          aria-label="Passengers"
          id="incrementerContainer"
          style={
                    Object {
                              "display": "none",
                            }
          }
>
          <BookPassengersSection
                    currentAmount={2}
                    decrementTabIndex={undefined}
                    disableIndividualSections={undefined}
                    id="Adults (18-64)"
                    index="0"
                    individualMaxLimit={undefined}
                    maxLimit={undefined}
                    maxRemaining={NaN}
                    minValueIsOne={undefined}
                    minusBtnDisabled={false}
                    modelName="booking.passengersToBook['Adults (18-64)']"
                    numberOfTravelersTotal={undefined}
                    onQuantityClick={[Function]}
                    onQuantityInputFocus={[Function]}
                    onTextInput={[Function]}
                    parentModel={undefined}
                    plusBtnDisabled={false}
                    sectionDetail="(18-64)"
                    sectionTitle="Adults"
          />
          <BookPassengersSection
                    currentAmount={0}
                    decrementTabIndex={undefined}
                    disableIndividualSections={undefined}
                    id="Seniors (65+)"
                    index="1"
                    individualMaxLimit={undefined}
                    maxLimit={undefined}
                    maxRemaining={NaN}
                    minValueIsOne={undefined}
                    minusBtnDisabled={false}
                    modelName="booking.passengersToBook['Seniors (65+)']"
                    numberOfTravelersTotal={undefined}
                    onQuantityClick={[Function]}
                    onQuantityInputFocus={[Function]}
                    onTextInput={[Function]}
                    parentModel={undefined}
                    plusBtnDisabled={false}
                    sectionDetail="(65+)"
                    sectionTitle="Seniors"
          />
          <BookPassengersSection
                    currentAmount={3}
                    decrementTabIndex={undefined}
                    disableIndividualSections={undefined}
                    id="Infants (under 2)"
                    index="2"
                    individualMaxLimit={undefined}
                    maxLimit={undefined}
                    maxRemaining={NaN}
                    minValueIsOne={undefined}
                    minusBtnDisabled={false}
                    modelName="booking.passengersToBook['Infants (under 2)']"
                    numberOfTravelersTotal={undefined}
                    onQuantityClick={[Function]}
                    onQuantityInputFocus={[Function]}
                    onTextInput={[Function]}
                    parentModel={undefined}
                    plusBtnDisabled={false}
                    sectionDetail="(under 2)"
                    sectionTitle="Infants"
          />
          <BookPassengersSection
                    currentAmount={1}
                    decrementTabIndex={undefined}
                    disableIndividualSections={undefined}
                    id="Infants on lap"
                    index="3"
                    individualMaxLimit={undefined}
                    maxLimit={undefined}
                    maxRemaining={NaN}
                    minValueIsOne={undefined}
                    minusBtnDisabled={false}
                    modelName="booking.passengersToBook['Infants on lap']"
                    numberOfTravelersTotal={undefined}
                    onQuantityClick={[Function]}
                    onQuantityInputFocus={[Function]}
                    onTextInput={[Function]}
                    parentModel={undefined}
                    plusBtnDisabled={false}
                    sectionDetail=""
                    sectionTitle="Infants on lap"
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BookPassengersMenu
    arrayIndexs={
        Array [
            "0",
            "1",
            "2",
            "3",
          ]
    }
    fullArray={
        Array [
            "Seniors (65+)",
            "Infants (under 2)",
            "Infants on lap",
          ]
    }
    ids={
        Array [
            "Adults (18-64)",
            "Seniors (65+)",
            "Infants (under 2)",
            "Infants on lap",
          ]
    }
    infantTitleA="Infants (under 2)"
    infantTitleB="Infants on lap"
    onChangeModel={[Function]}
    onChangePassenger={[Function]}
    onDisableTravelerButtons={[Function]}
    onQuantityClick={[Function]}
    onQuantityInputFocus={[Function]}
    passengersToBook={
        Immutable.Map {
            Adults (18-64): 2,
            Seniors (65+): 0,
            Infants (under 2): 3,
            Infants on lap: 1,
            Children (15-17): 0,
            Children (12-14): 0,
            Children (5-11): 0,
            Children (2-4): 0,
        }
    }
    sectionDetail={
        Array [
            "(18-64)",
            "(65+)",
            "(under 2)",
            "",
          ]
    }
    sectionTitle={
        Array [
            "Adults",
            "Seniors",
            "Infants",
            "Infants on lap",
          ]
    }
    toDisableBtns={
        Immutable.Map {
            plusBtn: false,
            minusBtn: false,
        }
    }
/>,
}
`;
